ethereum/contracts/EthereumApp.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./Decoder.sol";
+
+contract EthereumApp {
+    using SafeMath for uint256;
+    using Decoder for bytes;
+
+    uint256 public nonce;
+    uint256 public totalETH;
+
+    enum AppEventTags { SendETH, SendERC20 }
+
+    event AppEvent(uint _tag, bytes _data);
+    event Unlock(bytes _sender, address _recipient, uint256 _amount);
+
+    constructor() public {
+        nonce = 0;
+    }
+
+    function sendETH(
+        bytes32 _recipient
+    )
+        public
+        payable
+    {
+        require(msg.value > 0, "Value of transaction must be positive");
+
+        // Increment locked Ethereum counter by this amount
+        totalETH = totalETH.add(msg.value);
+        // Increment global nonce
+        nonce = nonce.add(1);
+
+        bytes memory data = encodeSendData(msg.sender, _recipient, address(0), msg.value, nonce);
+        emit AppEvent(uint(AppEventTags.SendETH), data);
+    }
+
+    function encodeSendData(
+        address _sender,
+        bytes32 _recipient,
+        address _tokenAddr,
+        uint256 _amount,
+        uint256 _nonce
+    )
+        internal
+        pure
+        returns(bytes memory)
+    {
+        return abi.encode(_sender, _recipient, _tokenAddr, _amount, _nonce);
+    }
+
+
+    function unlockETH(bytes memory _data)
```

Comment:
As with the ERC20 app we'll need to submit a signature too. And can we rename the function to `submit` to match what the relayer calls?

Reply:
So the method signature should be `submit(bytes memory _data, bytes memory _signature)`. Thoughts?

Reply:
I've made this change in the integration branch at https://github.com/Snowfork/polkadot-ethereum/pull/59

Reply:
See above